spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -319,11 +320,20 @@ else if (pageable instanceof DatastorePageable) {
 	private void applySelectWithFilter(Object[] parameters, Builder builder) {
 		Iterator it = Arrays.asList(parameters).iterator();
 		Filter[] filters = this.filterParts.stream().map((part) -> {
-			DatastorePersistentProperty persistentProperty = (DatastorePersistentProperty) this.datastorePersistentEntity
-					.getPersistentProperty(part.getProperty().getSegment());
+			Iterable<PropertyPath> iterable = () -> part.getProperty().iterator();
+			List<DatastorePersistentProperty> properties =
```

Comment:
Put this block into a separate helper private method `buildName` like you did for Firestore?